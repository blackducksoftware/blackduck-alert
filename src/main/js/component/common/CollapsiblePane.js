import React, { Component } from "react";
import PropTypes from "prop-types";

class CollapsiblePane extends Component {
    constructor(props) {
        super(props)
        this.state = { expanded: this.props.expanded }
        this.toggleDisplay = this.toggleDisplay.bind(this);
    }

    toggleDisplay() {
        this.setState({ expanded: !this.state.expanded })
    }

    render() {
        const contentClass = this.state.expanded ? 'shown' : 'hidden';
        return (
            <div className="collapsiblePane">
                <div className="title" onClick={this.toggleDisplay}>
                    {this.props.titleComponent}
                </div>
                <div className={contentClass}>
                    {this.props.children}
                </div>
            </div>
        );
    }
}

CollapsiblePane.propTypes = {
    titleComponent: PropTypes.object.isRequired,
    expanded: PropTypes.bool
};

CollapsiblePane.defaultProps = {
    titleComponent: '',
    expanded: false
};

export default CollapsiblePane;
