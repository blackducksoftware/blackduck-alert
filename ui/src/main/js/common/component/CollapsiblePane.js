import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const CollapsiblePane = ({ children, id, title, expanded }) => {
    const [isExpanded, setIsExpanded] = useState(expanded);

    function toggleCollapsiblePane() {
        setIsExpanded(!isExpanded);
    }

    return (
        <div className="collapsiblePanel">
            <button
                id={`${id}-expand-button`}
                type="button"
                className="btn btn-link"
                onClick={toggleCollapsiblePane}
            >
                <FontAwesomeIcon icon={isExpanded ? 'minus' : 'plus'} className="icon" size="lg" />
                {title}
            </button>
            <div className={isExpanded ? 'shown' : 'hidden'}>
                {children}
            </div>
        </div>
    );
};

CollapsiblePane.defaultProps = {
    id: 'collapsiblePaneId',
    expanded: false
};

CollapsiblePane.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.array,
        PropTypes.object
    ]),
    expanded: PropTypes.bool,
    id: PropTypes.string,
    title: PropTypes.string.isRequired
};

export default CollapsiblePane;
