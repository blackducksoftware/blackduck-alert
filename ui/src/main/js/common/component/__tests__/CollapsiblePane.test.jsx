import React from 'react';
import Enzyme  from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-15';
import CollapsiblePane from '../CollapsiblePane';

// TODO: Implement this icon library in a much more dynamic way - remove this when done
import registerFaIcons from '../../../icons';
registerFaIcons();

beforeAll(() => {
    Enzyme.configure({ adapter: new Adapter() });
});

const defaultChildren = ['Child 1', 'Child 2'];

test('Rendering default collapsible pane snapshot', () => {
    const component = renderer.create(<CollapsiblePane
        id={'collapsiblePaneId'}
        title={'collapsiblePane Configuration'}
        children={defaultChildren}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
