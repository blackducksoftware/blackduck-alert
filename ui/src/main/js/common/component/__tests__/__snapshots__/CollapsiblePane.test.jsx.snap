// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering default collapsible pane 1`] = `
<div
  className="collapsiblePanel"
>
  <button
    className="btn btn-link"
    id="collapsiblePaneId-expand-button"
    onClick={[Function]}
    type="button"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-plus fa-w-14 fa-lg icon"
      data-icon="plus"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    collapsiblePane Configuration
  </button>
  <div
    className="hidden"
  >
    Child 1
    Child 2
  </div>
</div>
`;
